test/ng/directive/ngEventDirsSpec.js

```diff
@@ -148,6 +148,163 @@ describe('event directives', function() {
       expect($rootScope.blur).toHaveBeenCalledOnce();
       expect(element.val()).toBe('newValue');
     }));
+  });
+
+  describe('events in digest phases', function() {
+
+    it('should not stop the digest when the handler throws an error', function() {
+
+      module(function($exceptionHandlerProvider) {
+        $exceptionHandlerProvider.mode('log');
+      });
+
+      inject(function($rootScope, $compile, $exceptionHandler, $log) {
+
+        element = $compile('<button ng-click="click()">Click</button>')($rootScope);
+        expect($log.assertEmpty());
+        $rootScope.click = function() {
+          throw new Error('listener error');
+        };
+
+        $rootScope.do = function() {
+          element.triggerHandler('click');
+          $log.log('done');
+        };
+
+        $rootScope.$apply(function() {
+          $rootScope.do();
+        });
+
+        expect($exceptionHandler.errors).toEqual([Error('listener error')]);
+        expect($log.log.logs).toEqual([['done']]);
+        $log.reset();
+      });
+    });
+
+
+    it('should not stop the digest when the handler throws an error', function() {
+
+      module(function($exceptionHandlerProvider) {
+        $exceptionHandlerProvider.mode('log');
+      });
+
+      inject(function($rootScope, $compile, $exceptionHandler, $log) {
+
+        element = $compile('<button ng-click="click()">Click</button>')($rootScope);
+        expect($log.assertEmpty());
+        $rootScope.click = function() {
+          throw new Error('listener error');
+        };
+
+        $rootScope.do = function() {
+          element.triggerHandler('click');
+          $log.log('done');
+        };
+
+        $rootScope.do();
+
+        expect($exceptionHandler.errors).toEqual([Error('listener error')]);
+        expect($log.log.logs).toEqual([['done']]);
+        $log.reset();
+      });
+    });
+
+    it('should not stop the digest when the handler is called in a watch expression fn', function() {
+
+      module(function($exceptionHandlerProvider) {
+        $exceptionHandlerProvider.mode('log');
+      });
+
+      inject(function($rootScope, $compile, $exceptionHandler, $log) {
+
+        element = $compile('<button ng-click="click()">Click</button>')($rootScope);
+        expect($log.assertEmpty());
+        $rootScope.click = function() {
+          throw new Error('listener error');
+        };
+
+        $rootScope.$watch(function() {
+          $rootScope.click();
+          $log.log('done');
+        });
+
+        $rootScope.$digest();
+
+        // The error gets caught in the traverseScopesLoop try catch, and the $log is never reached
+        expect($exceptionHandler.errors).toEqual([Error('listener error')]);
+        expect($log.log.logs).toEqual([]);
+        $log.reset();
+      });
+    });
+
+
+    it('should not stop the digest when the event is fired in a watch expression fn', function() {
```

Comment:
@jbedard if you trigger a click event handler inside a watch expression fn, then the additional try catch changes how the error gets logged, and in this case, anything that comes after the handler is called, whereas when you call the handler directly and throw, what comes after the handler is not called. See the test above for the different outcome.

Reply:
That is caused by the `click()` vs `triggerHandler('click')`? I didn't think that would change anything...?

Reply:
It's caused by the new try catch. To clarify, ij the watcher expression fn tests I test direct call to scope fn against event handler call to scope fn. The first test doesn't use the event handler at all. It's a bit of a digression from the original testing but it's an example where the new try catch leads to different behavior.

Reply:
Is the watch expression fn test set up like you intended it to be? Because you mentioned that you think such a test would make sense.

Reply:
Oh... I understand these two tests now. Sorry I was misreading `$rootScope.click()` as `$element.click()` for some reason (my fault 😅). I think this difference is ok, definitely worth the two `do()` tests behaving the same imo.

Reply:
I think this test is correct and good as-is 👍